src/uwtools/config/j2template.py

```diff
@@ -43,15 +43,15 @@ def __init__(
 
     # Public methods
 
-    def dump(self, output_path: DefinitePath) -> None:
+    def dump(self, output_path: OptionalPath) -> None:
         """
         Write rendered template to the path provided.
 
         :param output_path: Path to file to write.
         """
         msg = f"Writing rendered template to output file: {output_path}"
         logging.debug(msg)
-        with open(output_path, "w+", encoding="utf-8") as f:
+        with writable(output_path) as f:
```

Comment:
So, this is necessary to implement the rocoto piece you're trying for in this PR. I had assumed in our conversation before seeing the PR that you were handling the behavior of all the `uw template` functions, too, which I think would be out of scope for this work. 

I don't see a connection to the atparse bit, so I think that can be left off, unless I'm missing a connection. Def let me know if that's the case!

Reply:
That's how I ended up expanding initially, but this can be bypassed with converting to str earlier - I should have that in the most recent update

Reply:
It's not the technicality of passing a string or a Path I'm concerned with here. If the user wants to output to stdout instead of a file (the OptionalPath change), there is no way to achieve that without this writeable piece.

Reply:
Happily restored!